### PR TITLE
Support ER_CONSTRAINT_FAILED

### PR DESCRIPTION
**Pull request by pingz**
As said in Issue https://github.com/PyMySQL/mysqlclient-python/issues/263,

Already installed and tested this PR with mariadb 10.2.

### CODE REVIEW

File: MySQLdb/constants/CR.py [hunk 1]
```diff
@@ -102,3 +102,55 @@
 AUTH_PLUGIN_ERR = 2061
 MAX_ERROR = 2999
 ERROR_LAST = 2061
+MIN_ERROR = 2000
+UNKNOWN_ERROR = 2000
+SOCKET_CREATE_ERROR = 2001
```
Comment (methane) [inline MySQLdb/constants/CR.py]: Don't duplicate error definitions.

**Comment by methane** [comment]
Have you checked error codes for MySQL and MariaDB are consisntent?

File: MySQLdb/constants/CR.py (hunk 1 above)
Comment (pingz) [reply MySQLdb/constants/CR.py]: `CR_MIN_ERROR` and `CR_MAX_ERROR` needed be excluded in here, and same with `ER_ERROR_FIRST*` and `ER_ERROR_LAST*`.

**Comment by pingz** [comment]
@methane just compared two `mysqld_error.h` files, I have to say not exactly the same, just enough not to cause some serious conflict.

[ER-mariadb-10.2.patched.txt](https://github.com/PyMySQL/mysqlclient-python/files/2365842/ER-mariadb-10.2.patched.txt)
[ER-mysql-5.7.txt](https://github.com/PyMySQL/mysqlclient-python/files/2365843/ER-mysql-5.7.txt)

**Comment by methane** [comment]
If so, I use MySQL's one always.
Please add only MariaDB-only sections, instead of regenerate all errors.

**Comment by pingz** [comment]
@methane should I still sync mysql 5.7, you see mysql 5.7 is also different with preview version, start from  `1885`. If I want added a MariaDB-only sections, I think I want to sync `ER.py` to mysql 5.7 first.

**Comment by pingz** [comment]
Hi, @methane , I recently just think maybe I can write a simple script for merging two different `mysqld_error.h` files to generated an `ER.py`, same for the `CR.py`. should I close this PR and have a new one.

Let me know what is your thoughts on this.

**Comment by methane** [comment]
I don't want accept MariaDB's errors unless it's really necessary...  It increase maintenance cost.
Is ER_CONSTRAINT_FAILED MariaDB specific error?

**Comment by pingz** [comment]
I think it is mariadb only thing. `ER_CONSTRAINT_FAILED` is basically for standard SQL syntax `CHECK CONSTRAINT`, something only mariadb 10.2 implemented for now, maybe mysql will support it later. And mysql 8.0 still not support it yet.

https://bugs.mysql.com/bug.php?id=3464

Anyway, it is not something very important, I'm closing this PR and https://github.com/PyMySQL/mysqlclient-python/issues/263 .

**Pull request closed without merging**